pkg/kube/kube_handler.go

```diff
@@ -773,21 +1034,15 @@ func (a *userHandler) ValidateAppRepository(appRepoBody io.ReadCloser, requestNa
 	if err != nil {
 		return nil, err
 	}
-	reqs, err := getRequests(appRepo, cli)
+	httpValidator, err := getValidators(appRepo, cli)
 	if err != nil {
 		return nil, err
 	}
-	response := &ValidationResponse{}
-	for _, req := range reqs {
-		response, err = doValidationRequest(cli, req)
-		if err != nil {
-			return nil, err
-		}
-		if response.Code != 200 {
-			return response, nil
-		}
+	response, err := httpValidator.IsValid(cli)
+	if err != nil {
+		return nil, err
 	}
-	return response, nil
+	return response, err
```

Comment:
Why this change? It's easier to reason about when you know it's nil rather than err (and you've just returned the error if it was non-nil).